packages/after/lib/loadInitialProps.js

```diff
@@ -1,12 +1,11 @@
 import { matchPath } from 'react-router-dom';
 
 export default function loadInitialProps(routes, pathname, ctx) {
-    const match = matchPath(pathname, route);
-
   return Promise.all(
     routes
-      .filter(route => match && route.component.getInitialProps))
-      .map(route => route.component.load
+      .map(route => ({...route, match: matchPath(pathname, route) }))
```

Comment:
hmm....not sure this ends up being "cleaner." This will produce more code than the current implementation after it runs through Babel. 

I wonder if there is an even better way with just map and reduce? The array is only ever of length 1 or 0.

Reply:
completely agree with you. at first it seemed better but now that i see how it works, that seems like a better way